portal-api/src/v1/gef/services/clone-deal.service.test.ts

```diff
@@ -0,0 +1,210 @@
+import {
+  aPortalSessionUser,
+  Bank,
+  Deal,
+  DEAL_STATUS,
+  DealNotFoundError,
+  getCurrentGefDealVersion,
+  InvalidDealIdError,
+  InvalidParameterError,
+} from '@ukef/dtfs2-common';
+import { Collection, ObjectId } from 'mongodb';
+import { generateAuditDatabaseRecordFromAuditDetails, generateSystemAuditDetails } from '@ukef/dtfs2-common/change-stream';
+import { cloneDealToLatestVersion } from './clone-deal.service';
+import { mongoDbClient } from '../../../drivers/db-client';
+import { cloneExporter } from './clone-exporter.service';
+
+const mockLatestEligibilityCriteria = {
+  eligibility: 'criteria',
+};
+
+jest.mock('../controllers/eligibilityCriteria.controller', () => ({
+  getLatestEligibilityCriteria: jest.fn(() => mockLatestEligibilityCriteria),
+}));
+
+const existingDealId = new ObjectId().toString();
+const mockAuditDetails = generateSystemAuditDetails();
+const mockBank = {
+  id: 'bank id',
+} as Bank;
+const bankInternalRefName = 'bankInternalRefName';
+const additionalRefName = 'additionalRefName';
+const maker = aPortalSessionUser();
+
+const existingDeal: Deal = {
+  _id: new ObjectId(existingDealId),
+  version: 0,
+  dealType: 'GEF',
+  ukefDealId: 'UkefDealId',
+  eligibility: {},
+  exporter: {},
+  createdAt: Date.now(),
+  updatedAt: Date.now(),
+  facilitiesUpdated: Date.now(),
+  status: DEAL_STATUS.UKEF_ACKNOWLEDGED,
+  submissionType: null,
+  submissionDate: null,
+  submissionCount: 0,
+  bankInternalRefName: 'oldBankInternalRefName',
+  additionalRefName: 'oldAdditionalRefName',
+  maker: {},
+  bank: {},
+  checkerId: 'checker id',
+  editedBy: ['a user id', 'another user id'],
+  portalActivities: ['An activity'],
+  supportingInformation: {
+    information: 'information',
+  },
+  auditRecord: {},
+  ukefDecision: 'ukefDecision',
+  ukefDecisionAccepted: 'ukefDecisionAccepted',
+  checkerMIN: 'checkerMIN',
+  manualInclusionNoticeSubmissionDate: 'manualInclusionNoticeSubmissionDate',
+  comments: 'comments',
+  previousStatus: 'previousStatus',
+  dataMigration: 'dataMigration',
+};
```

Comment:
Can we use a mock/fixture? Rather than ~30 lines of mocks in the test.

Reply:
I thought a lot about this & I dont think it's ideal to do that - the tests overwrite ~25 properties on the deals so we really care about what value they were initially and what value they end up being.

I think if this is hidden by `const existingDeal = MOCK_GEF_DEAL`  then its a bit less intentional that we want, for example, the `dataMigration` value to be removed.

If I went down the route of
```
const existingDeal = {
  ...MOCK_GEF_DEAL,
  dataMigration: 'dataMigration',
  // etc...
}
```
Then I'd still need to add 25 values to check that they are being updated/overwritten

I also cant actually see any mock gef deal constants or helpers in portal-api

Reply:
Thanks @MarRobSoftwire , seems it makes sense on the deal level at least